From 9f6f266852382b77fb359675735a40203f8e400c Mon Sep 17 00:00:00 2001
From: Hsieh-Tseng Shen <woodrow.shen@canonical.com>
Date: Mon, 1 Jun 2020 21:10:51 +0800
Subject: [PATCH 1/6] Change policy rule to force to use sudo as default group

---
 policy/org.freedesktop.fwupd.rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/policy/org.freedesktop.fwupd.rules b/policy/org.freedesktop.fwupd.rules
index 651d4e35..cb04cdd2 100644
--- a/policy/org.freedesktop.fwupd.rules
+++ b/policy/org.freedesktop.fwupd.rules
@@ -1,7 +1,7 @@
 polkit.addRule(function(action, subject) {
     if (action.id == "org.freedesktop.fwupd.update-internal" &&
         subject.active == true && subject.local == true &&
-        subject.isInGroup("wheel")) {
+        subject.isInGroup("sudo")) {
             return polkit.Result.YES;
     }
 });
-- 
2.25.1

